dev/com.ibm.ws.microprofile.config.1.3/src/com/ibm/ws/microprofile/config13/sources/OSGiConfigUtils.java

```diff
@@ -213,8 +213,11 @@ public static String getApplicationConfigFilter(BundleContext bundleContext, Str
             String applicationFilter = getApplicationConfigFilter(bundleContext, applicationName);
             ConfigurationAdmin admin = getConfigurationAdmin(bundleContext);
             Configuration[] osgiConfigs = admin.listConfigurations(applicationFilter);
-            for (Configuration cfg : osgiConfigs) {
-                configSet.add(cfg);
+
+            if (osgiConfigs != null) {
```

Comment:
huh, didn't realise that it was possible for this to be null ... the check doesn't hurt though

Reply:
The check is required. In the absence of appProperties attributes in the server.xml application element I see NPEs.